scripts/src/java/org/oppia/android/scripts/todo/TodoCollector.kt

```diff
@@ -0,0 +1,47 @@
+package org.oppia.android.scripts.todo
+
+import org.oppia.android.scripts.common.RepositoryFile
+import org.oppia.android.scripts.todo.data.Todo
+
+/** Collects code lines containing the 'todo' keyword (case-insensitive). */
+class TodoCollector {
+  companion object {
+    private val todoDetectorRegex = Regex(pattern = "\\bTODO\\b", option = RegexOption.IGNORE_CASE)
+
+    /**
+     * Collects all the TODOs in the repository.
+     *
+     * @param repoPath the path to the repository
+     * @return a list of all the Todos
+     */
+    fun collectTodos(repoPath: String): List<Todo> {
+      val searchFiles = RepositoryFile.collectSearchFiles(repoPath)
+      return searchFiles.flatMap { file ->
+        file.bufferedReader()
+          .lineSequence()
+          .mapIndexedNotNull { lineIndex, lineContent ->
+            checkIfContainsTodo(
+              filePath = file.toString(),
+              lineContent = lineContent,
+              lineIndex = lineIndex
+            )
+          }
+      }
+    }
+
+    /**
+     * Computes whether a line of code contains the 'todo' keyword.
+     *
+     * @param filePath the path of the file
+     * @param lineContent the line string
+     * @param lineIndex the index of the line sequence which is to be searched for a TODO
+     * @return a Todo instance if the todo detector regex matches, else returns null
```

Comment:
Will this correctly let us detect malformed TODOs & print corresponding errors?

Reply:
Yes, this will perfectly detect the malformed TODOs too and we are printing the poor formatting error logs too.